### DISCUSSION TITLE
Should this Addon work with Tesla Powerwalls?

### DISCUSSION THREAD

**Discussion by barryperfect** [llamafilm/tesla-http-proxy-addon#43] [General]
I am trying use this addon to make my Tesla Custom Integration work again since Tesla moved me to the Fleet API.

Should using this add on work with Tesla Powerwalls as well as cars?

The reason I ask is that this step in the setup:

"Using iOS or Android Home Assistant Companion app, navigate to this add-on, select Web UI and click Generate OAuth Token. This will launch a web browser where you authenticate with Tesla. "

Only asks for vehicle permissions, whilst on the Tesla Developer Website I have enabled "Energy Product Information" and "Energy Product Commands" but this addon doesn't seem to request them?

I used to be able to view and control my Powerwall using Tesla Custom Integration before the API change.

**Comment by llamafilm** [comment]
Currently this addon only requests scopes for vehicles. I guess it could be modified to include energy sites but I haven’t tried.

**Reply by barryperfect** [reply]
It would be great if it could be made to work. When Tesla do finally kill off the Owner API I suspect there will be quite a few Powerwall owners looking for a plan B 😀

**Reply by llamafilm** [reply]
Your welcome to make a PR.

**Reply by llamafilm** [reply]
I added this in ba812bb163407280a1e0875ef0bae51fa3e3d2bb but I don't have a Powerwall to test.  Let me know if it works for you.
